lib/expand/builder.go

```diff
@@ -258,15 +257,19 @@ func (b *planBuilder) AddPostHookPhase(plan *storage.OperationPlan) {
 
 // AddElectPhase appends phase that enables leader election to the plan
 func (b *planBuilder) AddElectPhase(plan *storage.OperationPlan) {
-	plan.Phases = append(plan.Phases, storage.OperationPhase{
+	phase := storage.OperationPhase{
 		ID:          ElectPhase,
 		Description: "Enable leader election on the joined node",
 		Data: &storage.OperationPhaseData{
 			Server:     &b.JoiningNode,
 			ExecServer: &b.JoiningNode,
 		},
 		Requires: []string{installphases.WaitPhase},
-	})
+	}
+	if !b.JoiningNode.IsMaster() {
```

Comment:
We need to reset the election state (and not just enable it for masters) considering the use-case when a user might be reusing a node for a regular role that previously was a master.